src/MeshManager.cpp

```diff
@@ -268,6 +286,104 @@ namespace blitzdg {
         }
     }
 
+    void MeshManager::buildConnectivity() {
+        index_type totalFaces = NumFaces * NumElements;
+        index_type numVertices = NumVerts;
+        index_type localVertNum[3][2] = {{0, 1}, {1, 2}, {2, 0}};
+
+        // Build global face-to-vertex array. Note: we actually
+        // build its transpose for easy column access.
+        CSCMat VToF(numVertices, totalFaces, 2*totalFaces);
+        const index_vector_type& E2V = *EToV;
+        index_type globalFaceNum = 0;
+        index_type nnz=0;
+        for (index_type k = 0; k < NumElements; ++k) {
+            for (index_type f = 0; f < NumFaces; ++f) {
+                VToF.colPtrs(globalFaceNum) = nnz; 
+                index_type v1 = localVertNum[f][0];
+                index_type v2 = localVertNum[f][1];
+
+                index_type v1Global = E2V(k*NumFaces + v1);
+                index_type v2Global = E2V(k*NumFaces + v2);
+
+                // Entry for v1Global.
+                VToF.rowInds(nnz) = v1Global;
+                VToF.elems(nnz) = 1.0;
+                ++nnz;
+
+                // Entry for v2Global
+                VToF.rowInds(nnz) = v2Global;
+                VToF.elems(nnz) = 1.0;
+                ++nnz;
+
+                ++globalFaceNum;
+            }
+        }
+        VToF.colPtrs(totalFaces) = nnz;
+        CSCMat FToF = multiply(transpose(VToF), VToF);
+
+        // Count the number of face-to-face connections.
+        index_type connectionsCount = 0;
+        for (index_type j = 0; j < totalFaces; ++j) {
+            for (index_type k = FToF.colPtrs(j); k < FToF.colPtrs(j + 1); ++k) {
+                index_type i = FToF.rowInds(k);
+                if (i != j && std::abs(FToF.elems(k) - 2.0) < NodeTol) {
+                    ++connectionsCount;
+                }
+            }
+        }
+
+        index_vector_type e1(connectionsCount);
+        index_vector_type e2(connectionsCount);
+        index_vector_type f1(connectionsCount);
+        index_vector_type f2(connectionsCount);
+
+        f1 = 0;
+        f2 = 0;
+        e1 = 0;
+        e2 = 0;
+
+        // store the faces in each connection.
+        index_type c = 0;
+        for (index_type j = 0; j < totalFaces; ++j) {
+            for (index_type k = FToF.colPtrs(j); k < FToF.colPtrs(j + 1); ++k) {
+                index_type i = FToF.rowInds(k);
+                if (i != j && std::abs(FToF.elems(k) - 2.0) < NodeTol) {
+                    f1(c) = i;
+                    f2(c) = j;
+                    ++c;
+                }
+            }
+        }
+
+        // Convert from global face number to element number with local face number.
+        e1 = floor(f1 / NumFaces);
+        f1 = (f1 % NumFaces);
+        e2 = floor(f2 / NumFaces);
```

Comment:
Do you need the floor function here? Since this is integer division, it should truncate automatically.